doc/Requirements.md

```diff
@@ -0,0 +1,184 @@
+
+PF instances MUST not share any muttable data.
+(Why: to guarantee that different PF instances will not impact each others.)
+
+# Parameters
+
+A PF MUST be able to handle parameters.
+(Why: because the PF aims to abstract hardware and model it by parameters.)
+
+## Definitions
+
+hardware: system controlled by the PF. Not necessary material system.
+          This term was chosen because:
+           - historically the PF reference implementation was used to abstract hardware
+           - the subsystem term would arguably fit best is already used.
+(FIXME: choose subsystem instead, it is not used in fact.)
+
+## Value
+
+A parameter MUST have a value.
+(Why: because a parameter without value would not abstract any hardware.)
+
+A PF MUST support mutable parameters.
+(Why: To control the underlined hardware.)
+
+A PF MAY support immutable parameters, ie parameters which value is determined on start than read only.
+(Why: To permit hardware read only value reflection.)
+(Note: This is not implemented in the PF reference implementation.)
+
+This value MUST be gettable and settable.
+(Why: TODO)
+
+A value MUST have a data type. A data type designate a parameter invariants.
+(Why: TODO)
+
+A value MUST be convertible to a string.
+(Why: TODO)
+
+A bijection string to/from value SHOULD exist.
+(Why: TODO)
+
+### Data type
+
+A value data type defines the value properties:
+ - memory layout
+ - value constrains
+
+A PF SHOULD support the following types:
+ - fix point parameter Qm.n and UQm.n
+   (Why: ?)
+ - Signed (2's complement) and unsigned integer parameters
+   (Why: )
+ - Floating point IEEE 754
+   (Why: )
+ - String: null terminated fixed maximum size array of character
+   (Why: )
+ - Bit field following the C abi.
+   (Why: )
+
+## Identifiers
+Every parameter MUST have a identifier that uniquely identifies it.
+(Why: to identify a parameter outside the framework)
+
+This identifier SHOULD be a string.
+(Why: so that the (human) user can identify a parameter with ease.)
+
+Two PF with the same parameters MUST have the same identifier for those parameters.
+(Why: Persistence of parameter identifier across PF instances with the same configuration.
+      To identify parameters independently of the host machine and PF instance)
+
+### Parameter tree
+A parameter SHOULD be structured in a tree. Each parameter being a distinct tree leaf.
+(Why: Tree is a simple data structure that can be easily represented and is enough to map
+      underlined layers.)
+
+Each node of the tree SHOULD have it's own identifier with the same
+characteristics (type, independence, ...) than a parameter.
+(Why: To represent the tree without treating the leaf nodes specifically.)
+
+The identifier of each node of the tree SHOULD be a combination of it's parents.
+More specifically, if the identifier is a string it should be formated the same
+way as a file system path. Eg the parameter named "Root/child1/4/parameter1".
+(Why: Usual syntax to address trees.)
+
+
```

Comment:
take virtual parameters into account ? e.g. "a parameter SHOULD be mapped by a syncer; a parameter that is not mapped by a syncer is called a "virtual" parameter."